### DISCUSSION TITLE
Best way of dealing with Notification Intent name changes in 3.x

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1442] [Q&A] [answered]
I am assumning other teams have come up against this issue during migration. We have a server side Notifications API that our server side components can call to get Push Noitrfcations sent to devices.

When building a payload for android we include the `click_action` field and set it to "SHINY_NOTIFICATION_CLICK". That works for current clients using Shiny 2.7. However we are currently testing a new release that uses 3.2.4. That wants us to register the following intents

```
 [IntentFilter(
             new[]
             {
                  Shiny.ShinyNotificationIntents.NotificationClickAction,
                  ShinyPushIntents.NotificationClickAction
             },
```

Those values are "SHINY_LOCAL_NOTIFICATION_CLICK" and "SHINY_PUSH_NOTIFICATION_CLICK". With our server side component  sending a click_action of "SHINY_NOTIFICATION_CLICK" we have an issue that means the notification tap will not load the app.

~~I could just add the "SHINY_NOTIFICATION_CLICK" to the array of Intent Filters but I'm assuming that Shiny is no longer expecting that click action. It does sort of working but the instance of the app that appears looks like a new one rather than the existing one.~~

Useless idea anyway because that does not fix old clients that have not upgraded the app which is the main issue. We don't have a way of forcing people to upgrade (it is "in the backlog")

Adding multiple click_actions to the payload is not allowed. 

What did other people do?

**Comment by aritchie** [comment, marked as answer]
You can change the notification click action that push responds on through the AddPush configuration.  That might fix your situation.

**Reply by munkii** [reply]
Thanks Allan that’s worked.
